### DISCUSSION TITLE
Nuxt Modules | Module configuration file

### DISCUSSION THREAD

**Discussion by localusercamp** [nuxt/nuxt#16081] [Questions]
Hello. 

I want the module to be configurable via config file in project root like `nuxt.config.ts` (`mymodule.config.ts` for example);

I saw [tailwindcss solution](https://github.com/nuxt-modules/tailwindcss/blob/main/src/module.ts). But nuxt marked `resolveModule`, `requireModule`, `importModule` and so on as deprecated. Can i use them or there are some other solutions?

It will be nice if `mymodule.config.ts` could use `defineMyModuleConfig()` function which i also can provide with module, for better typehints.

How can i do that?

**Comment by wobsoriano** [comment]
Try https://github.com/unjs/c12

**Reply by localusercamp** [reply]
> Try https://github.com/unjs/c12

Does it support defining configuration via a function call?
For example: 
![image](https://user-images.githubusercontent.com/45385480/209288639-2cf23896-2906-4deb-92b0-1ab297a83208.png)

**Reply by localusercamp** [reply]
I mean how can i define a config with `defineMyModuleConfig` function if module autoimport that function

**Reply by localusercamp** [reply]
![image](https://user-images.githubusercontent.com/45385480/209290703-e0d9d191-e96e-4065-a533-3aeccd311529.png)
